test/libsolidity/util/TestFileParser.h

```diff
@@ -0,0 +1,195 @@
+/*
+	This file is part of solidity.
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <libdevcore/CommonData.h>
+#include <libsolidity/ast/Types.h>
+#include <liblangutil/Exceptions.h>
+
+#include <iosfwd>
+#include <stdexcept>
+#include <string>
+#include <vector>
+#include <utility>
+
+namespace dev
+{
+namespace solidity
+{
+namespace test
+{
+
+/**
+ * Format information used for the conversion of human-readable
+ * function arguments and return values to `bytes`. Defaults to
+ * a 32-byte representation.
+ */
+struct ByteFormat
```

Comment:
Is the plan to have exactly one of these per ABI type or do you also plan to include "formatting information" like whether an integer is stored in hex or decimal? If not, then shouldn't this just be ABIType or something like that?